core/src/main/java/org/apache/calcite/util/format/postgresql/NumberFormatPattern.java

```diff
@@ -27,13 +27,35 @@
  * A format element that will produce a number. Numbers can have leading zeroes
  * removed and can have ordinal suffixes.
  */
-public class NumberFormatPattern implements FormatPattern {
-  private final String[] patterns;
+public class NumberFormatPattern extends FormatPattern {
+  private final ChronoUnitEnum chronoUnit;
+  private final long minValue;
+  private final long maxValue;
+  private final int preferredLength;
   private final Function<ZonedDateTime, String> converter;
+  private final @Nullable Function<Integer, Integer> valueAdjuster;
 
-  protected NumberFormatPattern(Function<ZonedDateTime, String> converter, String... patterns) {
+  public NumberFormatPattern(ChronoUnitEnum chronoUnit, long minValue, long maxValue,
+      int preferredLength, Function<ZonedDateTime, String> converter, String... patterns) {
```

Comment:
what is preferredLength?

Reply:
Some format patterns such as `YYYY` are usually 4 characters long, but can be shorter. `YYYY` can also be used to parse a year value of `56`. `preferredLength` is the expected number of characters that the pattern consume, but it may actually consume fewer.

Consider this:
```
to_date('56-03-15', 'YYYY-MM-DD')
```

Reply:
I think this would be a very useful comment in the code as well.

Reply:
Added more Javadoc comments and tried to explain this better in the Javadoc comment.